src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,56 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAX_CAPACITY = 10;
+    private Object[] keys;
+    private Object[] values;
+    private int size;
+
+    public StorageImpl() {
+        keys = new Object[MAX_CAPACITY];
+        values = new Object[MAX_CAPACITY];
+        size = 0;
+    }
+
     @Override
     public void put(K key, V value) {
+        if (key == null) {
+            for (int i = 0; i < size; i++) {
+                if (keys[i] == null) {
+                    values[i] = value; // Replace the value if key is null
+                    return;
+                }
+            }
+        } else {
+            for (int i = 0; i < size; i++) {
+                if (key.equals(keys[i])) {
+                    values[i] = value; // Replace the value if key already exists
+                    return;
+                }
+            }
+        }
+
+        if (size < MAX_CAPACITY) {
+            keys[size] = key;
+            values[size] = value;
+            size++;
+        } else {
+            System.out.println("Storage is full, cannot add more elements.");
+        }
     }
 
     @Override
     public V get(K key) {
+        for (int i = 0; i < size; i++) {
```

Comment:
```suggestion
       int index = indexOf(key);
       return index == -1 ? null : values[index];
```

Also, it's better to create constant for -1 value, like NO_INDEX for example

Reply:
Done